### PR TITLE
add blosc and bloscpack args

### PR DESCRIPTION
**Pull request by mrocklin**
Add args for blosc and bloscpack

Turn off bloscpack offsets and checksum (I'm not sure that this is wise)

Specify blosc args per dtype.

### CODE REVIEW

**Comment by esc** [comment]
You probably also want to make the ``chunk_size`` configurable.

File: castra/core.py [hunk 1]
```diff
@@ -17,6 +17,18 @@
 from pandas import msgpack
 
 
+bp_args = bloscpack.BloscpackArgs(offsets=False, checksum='None')
+
+def blosc_args(dt):
```
Comment (esc) [inline castra/core.py]: Not sure about these defaults. Should experiment a bit to find good ones. Shuffle can work well on floats too.

Comment (mrocklin) [inline castra/core.py]: These defaults came about from experimentation on the nyctaxi dataset.  I've found that shuffle works well on integer valued data and very well on categorical or timestamp data but offers relatively slim gains on the floating point values (often lat/long).  

I've also generally found that with decent hard drives the costs of more intense compression quickly overwhelm the space saving benefits, leading me to prefer lower compression settings.

**Comment by mrocklin** [comment]
I'd like to merge this soon if there are no objections.

**Comment by esc** [comment]
I still think that the defaults are failry specific to the NYC-Taxi Dataset, so I'd object to those. But on the other hand there is no easy way to determine defaults. Maybe some kind of (dask-baed) hyperparameter search and cross-validation could be used to quasi-fit optimal parameters. (although I doubt that the high-dimensional parameter space will be convex and it is discrete)

**Comment by esc** [comment]
Anyway, let's merge this in for now and see what happens.

**Pull request merged**